src/runtime_src/core/pcie/linux/device_linux.cpp

```diff
@@ -51,6 +52,42 @@ struct bdf
   }
 };
 
+struct kds_custats
+{
+  using result_type = query::kds_custats::result_type;
+
+  static result_type
+  get(const xrt_core::device* device, key_type)
+  {
+    std::string errmsg;
```

Comment:
For code readability, it is sometimes best to define a variable just before it is needed.  In this case, the errmsg string should be moved down to line 83.